pkg/generic/mapthrift_codec_test.go

```diff
@@ -58,6 +59,42 @@ func TestMapThriftCodec(t *testing.T) {
 	test.Assert(t, err == nil)
 }
 
+func TestMapThriftCodecSelfRef(t *testing.T) {
```

Comment:
You could add the test using dynamicgo as well.

Reply:
dynamicgo doesn't support map generic-call, thus I add it into json-codec test